pkg/purl/purl.go

```diff
@@ -310,6 +313,9 @@ func parseMaven(pkgName string) (string, string) {
 
 // ref. https://github.com/package-url/purl-spec/blob/a748c36ad415c8aeffe2b8a4a5d8a50d16d6d85f/PURL-TYPES.rst#golang
 func parseGolang(pkgName string) (string, string) {
+	if strings.HasPrefix(pkgName, "./") || strings.HasPrefix(pkgName, "../") {
```

Comment:
We need a comment on why we skip these cases.

Reply:
Added